Tests/SwiftProtobufTests/Test_JSON.swift

```diff
@@ -970,7 +970,226 @@ final class Test_JSON: XCTestCase, PBTestHelpers {
+        + #""singularSfixed32":0,"#
+        + #""singularSfixed64":"0","#
+        + #""singularFloat":0.0,"#
+        + #""singularDouble":0.0,"#
+        + #""singularBool":false,"#
+        + #""singularString":"","#
+        + #""singularBytes":"","#
+        + #""singularNestedEnum":"FOO","#
+        + #""repeatedInt32":[],"#
+        + #""repeatedInt64":[],"#
+        + #""repeatedUint32":[],"#
+        + #""repeatedUint64":[],"#
+        + #""repeatedSint32":[],"#
+        + #""repeatedSint64":[],"#
+        + #""repeatedFixed32":[],"#
+        + #""repeatedFixed64":[],"#
+        + #""repeatedSfixed32":[],"#
+        + #""repeatedSfixed64":[],"#
+        + #""repeatedFloat":[],"#
+        + #""repeatedDouble":[],"#
+        + #""repeatedBool":[],"#
+        + #""repeatedString":[],"#
+        + #""repeatedBytes":[],"#
+        + #""repeatedNestedMessage":[],"#
+        + #""repeatedNestedEnum":[],"#
+        + #""mapInt32Int32":{},"#
+        + #""mapInt64Int64":{},"#
+        + #""mapUint32Uint32":{},"#
+        + #""mapUint64Uint64":{},"#
+        + #""mapSint32Sint32":{},"#
+        + #""mapSint64Sint64":{},"#
+        + #""mapFixed32Fixed32":{},"#
+        + #""mapFixed64Fixed64":{},"#
+        + #""mapSfixed32Sfixed32":{},"#
+        + #""mapSfixed64Sfixed64":{},"#
+        + #""mapInt32Float":{},"#
+        + #""mapInt32Double":{},"#
+        + #""mapBoolBool":{},"#
+        + #""mapStringString":{},"#
+        + #""mapStringBytes":{},"#
+        + #""mapStringNestedMessage":{},"#
+        + #""mapStringNestedEnum":{},"#
+        + #""repeatedBoolWrapper":[],"#
+        + #""repeatedInt32Wrapper":[],"#
+        + #""repeatedInt64Wrapper":[],"#
+        + #""repeatedUint32Wrapper":[],"#
+        + #""repeatedUint64Wrapper":[],"#
+        + #""repeatedFloatWrapper":[],"#
+        + #""repeatedDoubleWrapper":[],"#
+        + #""repeatedStringWrapper":[],"#
+        + #""repeatedBytesWrapper":[],"#
+        + #""optionalNullValue":null,"#
+        + #""repeatedDuration":[],"#
+        + #""repeatedTimestamp":[],"#
+        + #""repeatedFieldmask":[],"#
+        + #""repeatedAny":[],"#
+        + #""repeatedValue":[],"#
+        + #""repeatedListValue":[],"#
+        + #""repeatedStruct":[],"#
+        + #""fieldname1":0,"#
+        + #""fieldName2":0,"#
+        + #""FieldName3":0,"#
+        + #""fieldName4":0,"#
+        + #""field0name5":0,"#
+        + #""field0Name6":0,"#
+        + #""fieldName7":0,"#
+        + #""FieldName8":0,"#
+        + #""fieldName9":0,"#
+        + #""FieldName10":0,"#
+        + #""FIELDNAME11":0,"#
+        + #""FIELDName12":0,"#
+        + #""FieldName13":0,"#
+        + #""FieldName14":0,"#
+        + #""fieldName15":0,"#
+        + #""fieldName16":0,"#
+        + #""fieldName17":0,"#
+        + #""FieldName18":0"#
+        + "}"
+                
+        var encodingOptions = JSONEncodingOptions()
+        encodingOptions.alwaysPrintPrimitiveFields = true
+        assertJSONEncode(
```

Comment:
Since these don't always round trip (the configure == expected problem, it might be better/clearer to just skip the helpers changes and use do:


let msg = SwiftProtoTesting_Test3_TestAllTypesProto3()
let encoded = try msg.jsonString(options: encodingOptions)
XCTAssert(expected == encoded, "Did not encode correctly: got \(encoded) but expected \(expected)")

When when you do set some fields, use `with` when setting up `msg`.